src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -435,9 +441,17 @@ public override string ConnectionString
                     SqlConnectionString connectionOptions = new SqlConnectionString(value);
                     if (_credential != null)
                     {
+                        // Check for Credential being used with Authentication=ActiveDirectoryIntegrated. Since a different error string is used
+                        // for this case in ConnectionString setter vs in Credential setter, check for this error case before calling
+                        // CheckAndThrowOnInvalidCombinationOfConnectionStringAndSqlCredential, which is common to both setters.
+                        if (UsesActiveDirectoryIntegrated(connectionOptions))
+                        {
+                            throw SQL.SettingIntegratedWithCredential();
+                        }
+
                         CheckAndThrowOnInvalidCombinationOfConnectionStringAndSqlCredential(connectionOptions);
                     }
-                    else
+                    else if (_accessToken != null)
```

Comment:
This looked like an oversight in the netfx to netcore port. Minor unreported bug most likely.